sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,307 @@
+= SYCL task_sequence
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+[float]
+
+== Contributors
+
+// spell-checker: disable
+Jessica Davies, Intel +
+Joe Garvey, Intel +
+Robert Ho, Intel +
+Tommy Hoffner, Intel +
+Ajaykumar Kannan, Intel +
+Michael Kinsner, Intel +
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2021 Intel Corporation
+
+Intel Confidential
+
+== Status
+
+Working Draft
+
+This is a preview extension specification, intended to provide early access to a feature for review and community feedback. When the feature matures, this specification may be released as a formal extension.
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+
+== Version
+
+Built On: {docdate} +
+Revision: C
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, revision 3.
+
+== Overview
+
+Although SYCL supports task-level parallelism through device kernels, there is no specific mechanism to express parallelism at a sub-kernel level or to invoke additional kernels without involving the host program. In OpenCL 2.0 and beyond, kernel invocations can be initiated without host program interaction through device side enqueue. C++11 also introduced `std::async()` for task-level parallelism. However, neither of these approaches is a good fit for spatial compute platforms, which require a clear mapping of parallel constructs to spatial instances in the hardware.
+
+Rather than significantly changing the semantics of existing solutions, this extension introduces a sub-kernel task-level parallelism interface to SYCL for use in device code. It provides the class +task_sequence+, which is an abstraction of an asynchronous sequence of invocations of a callable that can be executed asynchronously from the caller, and with which a clear mapping to spatial hardware instances can be reasoned about when developing applications.
+
+== Definitions
+
+
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+template<auto& f>
+struct task_sequence{};
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          typename propertyListT = property_list<>>
```

Comment:
The latest version is called "properties" instead of "property_list".  Please update the name and the "task_sequence Properties" section to conform with the latest version of the extension.

Reply:
Updated.